Shared/src/com/pacoapp/paco/shared/model2/SQLQuery.java

```diff
@@ -108,10 +108,18 @@ public Builder fullEventAndOutputs(Boolean fullEventAndOutputs) {
     }
 
     private Builder addDefaultValues(SQLQuery obj) {
-        // provide default sort order which is Event._Id desc
-      if (obj.getGroupBy() == null && obj.sortOrder == null) {
-         obj.sortOrder = EventBaseColumns.TABLE_NAME+"."+Constants.UNDERSCORE_ID.concat(Constants.BLANK).concat(Constants.DESC);
-       }
+      // find if there is a distinct clause
+      boolean isDistinct = false;
+      for(String s : obj.getProjection()) {
+        if (s.startsWith(Constants.DISTINCT)) {
+          isDistinct = true;
+          break;
+        }
+      }
+      // provide default sort order which is Event._Id desc
+      if (obj.getGroupBy() == null && obj.sortOrder == null && !isDistinct) {
```

Comment:
micro-optimization: if it is the case that obj.getGroupBy != null || obj.sortOrder != null then we do not need to check the projections for a distinct. Is that correct? I imagine this is not expensive so I wouldn't change it just for this case, I just wanted to make sure that I understand it correctly.

Reply:
I am glad you asked the question. It was a great catch.